src/main/java/com/puppycrawl/tools/checkstyle/checks/design/DesignForExtensionCheck.java

```diff
@@ -322,7 +348,36 @@ private static boolean branchContainsJavadocComment(DetailAST token) {
             }
             curNode = toVisit;
         }
+        if (curNode != null) {
+            result = isValidJavadocComment(curNode, requiredJavadocPhrase);
```

Comment:
I did this only at the beginning, but there were survived mutations, so I changed it to this condition after analyzing.

Reply:
Oh I see, can you share which mutation it survived?

Reply:
Sorry, not mutation survival, but branch was not covered. This was because if the node was of not the type `BLOCK_COMMENT_BEGIN`, the `curNode` will stay null, and `curNode.getType() ` will throw a `NullPointerException`. Now, if we add an additional preceding condition of `if(curNode != null && ...)` then, one of the four branches survive and I found out that only keeping the `curNode != null` condition satisfies every condition since its either `null` or `BLOCK_COMMENT_BEGIN`.

Reply:
I tried the following implementation and it had not pitest failures or uncovered branches:
```
private static boolean branchContainsJavadocComment(DetailAST token,
                                                        Pattern requiredJavadocPhrase) {
        boolean result = false;
        DetailAST curNode = token;
        while (curNode != null) {
            if (curNode.getType() == TokenTypes.BLOCK_COMMENT_BEGIN
                    && JavadocUtil.isJavadocComment(curNode)) {
                result = isValidJavadocComment(curNode, requiredJavadocPhrase);
                break;
            }

            DetailAST toVisit = curNode.getFirstChild();
            while (toVisit == null) {
                if (curNode == token) {
                    break;
                }

                toVisit = curNode.getNextSibling();
                curNode = curNode.getParent();
            }
            curNode = toVisit;
        }
        return result;
    }
```
Please check

Reply:
Oh, sorry, I misunderstood your concern.